core/src/scene/stops.cpp

```diff
@@ -113,53 +113,29 @@ auto Stops::Sizes(const YAML::Node& _node, uint8_t _units) -> Stops {
         lastKey = key;
 
         if (frameNode[1].IsScalar()) {
-
             StyleParam::SizeValue sizeValue;
-            // default size values
-            sizeValue.fill({NAN, Unit::pixel});
-            bool validStop = constructFrame(frameNode[1], 0, sizeValue[0]);
-            if (!validStop) { continue; }
-
-            bool hasPercentage = sizeValue[0].isPercentage();
-
-            // Check for mixed dimension stops
-            if (dimSet && !hasSingleDim && !hasPercentage) {
-                LOGW("Can not have mixed dimensions stops for Size style parameter.");
-                stops.frames.clear();
-                return stops;
+            if (!constructFrame(frameNode[1], sizeValue.x)) {
+                continue;
             }
-
-            if (!dimSet && !hasPercentage) {
-                dimSet = true;
+            if (!sizeValue.x.isPercentage()) {
+                has1DSize = true;
             }
-
             stops.frames.emplace_back(key, sizeValue);
         } else if (frameNode[1].IsSequence()) {
-
-            // Check for mixed dimension stops
-            if (dimSet && hasSingleDim) {
-                LOGW("Can not have mixed dimensions stops for Size style parameter.");
-                stops.frames.clear();
-                return stops;
-            }
-
-            if (!dimSet) {
-                dimSet = true;
-                hasSingleDim = false;
-            }
-
             StyleParam::SizeValue sizeValue;
-            // default size values
-            sizeValue.fill({NAN, Unit::pixel});
             const auto& sequenceNode = frameNode[1];
-
-            bool validStop = constructFrame(sequenceNode[0], 0, sizeValue[0]);
-            if (!validStop || !constructFrame(sequenceNode[1], 0, sizeValue[1])) {
+            if (!constructFrame(sequenceNode[0], sizeValue.x) ||
+                !constructFrame(sequenceNode[1], sizeValue.y)) {
                 continue;
             }
-
+            has2DSize = true;
             stops.frames.emplace_back(key, sizeValue);
         }
+        if (has1DSize && has2DSize) {
```

Comment:
The difference here is the early exit. If one of the frames is "bad", we can save on some evaluations.